src/framework/global/uri.cpp

```diff
@@ -203,3 +203,10 @@ void UriQuery::addParam(const std::string& key, const Val& val)
 {
     m_params[key] = val;
 }
+
+UriQuery UriQuery::addingParam(const std::string& key, const Val& val) const
```

Comment:
I think it is better to use Uri Query::addParam directly where you need to add a param to URI. This method is a little bit unclear (at least for me :))